### PR TITLE
tests: drivers: build_all: Add tests for crypto, disk, and w1

### PR DESCRIPTION
**Pull request by soburi**
Add build_all test for crypto, disk, and w1 drivers.

- maxim,ds2482-800
- maxim,ds2484
- maxim,ds2485
- zephyr,sdmmc-disk
- zephyr,mmc-disk
- atmel,ataes132a

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]
Please add MAINTAINER.yml entries for the added tests.

**Comment by soburi** [comment]
@henrikbrixandersen 

> Please add MAINTAINER.yml entries for the added tests.

There is one problem.

The `misc` area is not defined in MAINTAINERS.yaml.

The subdirectories of drivers/misc are separated and handled in each area, but the 'misc' domain is not defined.

Since the necessary knowledge cannot be standardized, assigning a maintainer will be difficult, but what do you think is the best solution?


The others of `misc`' are already put.

**Comment by henrikbrixandersen** [comment]
> @henrikbrixandersen
> 
> > Please add MAINTAINER.yml entries for the added tests.
> 
> There is one problem.
> 
> The `misc` area is not defined in MAINTAINERS.yaml.
> 
> The subdirectories of drivers/misc are separated and handled in each area, but the 'misc' area is not defined.
> 
> Since the necessary knowledge cannot be standardized, assigning a maintainer will be difficult, but what do you think is the best solution?
> 
> The others of `misc`' are already put.

I'd leave out the entry for `misc` from this PR, then.

**Comment by soburi** [comment]
@henrikbrixandersen 

> I'd leave out the entry for `misc` from this PR, then.

OK, I'll deal with that way for now.

**Review by MaureenHelm — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Comment by aescolar** [comment]
From a quick glance I don't see any problem with this PR, but I add as reviewers the respective areas maintainers and collaborators so they can take a look

File: tests/drivers/build_all/disk/app.overlay [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024 TOKITA Hiroshi
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	test {
+		#address-cells = <1>;
```
Comment (danieldegrasse) [inline tests/drivers/build_all/disk/app.overlay:8]: There are more disk devices we should add here- we can definitely build the ramdisk driver here, and I suspect we can also build the flash disk driver for native_sim, right?

Comment (soburi) [reply tests/drivers/build_all/disk/app.overlay:8]: I got.
I will address this in another PR.

**Pull request merged**